src/Compilers/CSharp/Portable/Symbols/ReducedExtensionMethodSymbol.cs

```diff
@@ -103,10 +102,7 @@ internal override MethodSymbol CallsiteReducedFromMethod
 
         public override TypeSymbol ReceiverType
         {
-            get
-            {
-                return _reducedFrom.Parameters[0].Type;
-            }
+            get { return _typeMap.SubstituteType(_reducedFrom.Parameters[0].Type).Type; }
```

Comment:
Thanks for fixing this. Please port this change to master, along with a corresponding test. Or alternatively log an issue.